2020/day-12/arora-aditya/day12.py

```diff
@@ -0,0 +1,113 @@
+from typing import List, Dict, Set
+from collections import *
+from functools import lru_cache
+from math import log10
+from math import ceil
+from helper.submit.submit import *
+from utils import *
+
+
+DAY = 12
+setup(DAY)
+lines = read_file(DAY)
+
+F = "E"
+
+# E N
+e = 0
+n = 0
+
+def process(dir, count):
+    if dir == "N":
+        return 0, count
+    elif dir == "S":
+        return 0, -1 * count
+    elif dir == "E":
+        return count, 0
+    elif dir == "W":
+        return -1*count , 0
+
+order = list("NESW")
+
+def move(line):
+    global F
+    global e
+    global n
+    dir, count = line[0], int(line[1:])
+    if dir in set("NEWS"):
+        de, dn = process(dir, count)
+        e += de
+        n += dn
+    elif dir == "F":
+        de, dn = process(F, count)
+        e += de
+        n += dn
+    elif dir == "L":
+        if count % 90 != 0:
+            sys.exit(1)
+        for i in range(ceil(count/90)):
+            F = order[order.index(F)-1]
+    elif dir == "R":
+        if count % 90 != 0:
+            sys.exit(1)
+        for i in range(ceil(count/90)):
+            F = order[(order.index(F)+1)%len(order)]
```

Comment:
damn nice solution

Another student here did this, which took me a sec to get, but is actually pretty nifty

```python
def rotate_waypoint(x, y, r):
    count = r // 90
    for _ in range(count):
        x, y = y, x
        x *= -1
    return x, y
```

Reply:
that is indeed nifty